sled-agent/config-reconciler/src/reconciler_task/zones.rs

```diff
@@ -109,64 +130,55 @@ impl OmicronZones {
         // Filter desired zones down to just those that we need to stop. See
         // [`ZoneState`] for more discussion of why we're willing (or unwilling)
         // to stop zones in various current states.
-        let mut zones_to_shut_down = self
```

Comment:
This change (which is mostly whitespace) and the almost-identical change in `start_zones_if_needed` are both to workaround a rustc bug in the https://github.com/rust-lang/rust/issues/110338 family. `peek()` seems to trigger it [similarly to `flatten()`](https://github.com/rust-lang/rust/issues/126044).